linera-sdk/src/contract/runtime.rs

```diff
@@ -33,11 +33,9 @@ where
     application_id: Option<ApplicationId<Application::Abi>>,
```

Comment:
How about `test_runtime.rs`?

Reply:
There is caching in the integration tests.
I could reproduce the bug there.

Reply:
Yes, this makes sense the integration tests are using the real Wasm VM. `test_runtime.rs` is for unit tests